docs/components/file-upload.md

```diff
@@ -0,0 +1,310 @@
+  </template>
+</KCard>
+
+```html
+<KFileUpload
+  :removable=false
+  label="Upload File" 
+  :label-attributes="{ help: 'Accepted file types: .yml, .yaml, .json, .md, .markdown, image/*' }"
+  buttonAppearance="creation" 
+/>
+```
+
+<KCard class="mt-6">
+  <template v-slot:body>
+    <KFileUpload type="image" label="Upload File" :label-attributes="{ help: `Accepted file types: ${acceptedImageType}` }" :removable=false class="image-with-label" :accept="acceptedImageType" placeholder="No cancel button!" icon="kong" >
+    </KFileUpload>
+  </template>
+</KCard>
+
+```html
+<KFileUpload 
+  type="image"
+  :removable=false
+  label="Upload File"
+  placeholder="No cancel button!" 
+  icon="kong"
+/>
+```
+
+### accept
+
+Please refer to [mdn web docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#accept) that defines the file types the file input should accept. 
+
+### buttonAppearance
+
+Use this prop to customize the trigger **KButton** [appearance](/components/button.html#appearance). This button only shows when prop `type` is `file`. Default vaue is set to `primary`.
+
+<KCard>
+  <template v-slot:body>
+    <KFileUpload label="Upload File" :label-attributes="{ help: `Accepted file types: ${acceptedFileType}` }" buttonAppearance="creation" :accept="acceptedFileType" />
+  </template>
+</KCard>
+
+```html
+<KFileUpload label="Upload File" buttonAppearance="creation" />
+```
+
+### buttonText
+
+This is the text that will be displayed on the button that triggers the click on KInput. Button text can only be changed when prop `type` is `file`.
+
+<KCard>
+  <template v-slot:body>
+    <KFileUpload label="Upload File" :label-attributes="{ help: `Accepted file types: ${acceptedFileType}` }" buttonAppearance="danger" buttonText="Click me" :accept="acceptedFileType" />
+  </template>
+</KCard>
+
+```html
+<KFileUpload label="Upload File" buttonAppearance="danger" buttonText="Click me" />
+```
+
+### icon
+
+Specify an icon to display to the left of the `placeholder text` if prop `type` is `image`.
+
+### iconSize
+
+The size of the `icon` being displayed (default is `26`).
+
+### iconColor
+
+The color of the `icon` being displayed.
+
+<KCard class="mt-6">
+  <template v-slot:body>
+    <KFileUpload type="image" label="Upload File" :label-attributes="{ help: `Accepted file types: ${acceptedImageType}` }" :accept="acceptedImageType" class="image-with-label" placeholder="Customized icon, iconColor & iconSize!" icon="immunity" iconColor="gold" iconSize="30" />
+  </template>
+</KCard>
+
+```html
+<KFileUpload 
+  type="image" icon="immunity" iconColor="gold" iconSize="30" 
+  placeholder="You can change the text here!"
+/>
+```
+### maxFileSize
+
+Use this prop to customize the maximize size of file that can be uploaded. Default value for `image` upload is `1mb (1000000 bytes)` and for `file` is `5.24mb (5242880 bytes)`.
```

Comment:
nit: I don't think we need to call out the size conversion, and the abbreviation should be uppercase.

Also, why `5.24MB` for files? That seems very specific; why not just `5MB`?

```suggestion
Use this prop to customize the maximize size of file that can be uploaded. Default value for `image` upload is `1MB` and for `file` is `5.24MB`.
```

Reply:
`khcp-ui` `DocUploadCard` uses 5.24MB as max file size; that's why I selected that as the max size.